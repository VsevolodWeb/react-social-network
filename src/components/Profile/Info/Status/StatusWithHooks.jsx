import React, { useState } from 'react';
import s from './Status.module.css'

const StatusWithHooks = props => {

    const [editMode, setEditMode] = useState(false);
    const [status, setStatus] = useState(props.status);

    const toggleEditMode = () => {
        setEditMode(!editMode);

        if(props.status !== status) {
            props.updateUserStatus(status);
        }
    }

    const statusChange = e => {
        setStatus(e.currentTarget.value);
    }

    return (
        <div className={s.container}>
            {props.editingAbility ?
                editMode ?
                    <input type="text" onBlur={toggleEditMode} onChange={statusChange} value={status} autoFocus />
                    : <span className={`${s.status} ${s.status_editingAbility}`} onDoubleClick={toggleEditMode}>{status}</span>
                : <span className={s.status}>{status}</span>
            }
        </div>
    )
}

export default StatusWithHooks;