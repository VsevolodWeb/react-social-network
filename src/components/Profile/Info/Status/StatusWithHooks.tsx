import React, {ChangeEvent, useEffect, useState} from 'react'
import {useDispatch} from 'react-redux'
import {profileAPI} from '../../../../api/profile-api'
import {actions} from '../../../../redux/profile-reducer'
import s from './Status.module.css'

type PropsType = {
    status: string
    editingAbility: boolean
}

const StatusWithHooks: React.FC<PropsType> = props => {
    const [editMode, setEditMode] = useState(false)
    const [status, setStatus] = useState(props.status)
    const dispatch = useDispatch()

    useEffect(() => {
        setStatus(props.status)
    }, [props.status])

    const toggleEditMode = async () => {
        setEditMode(!editMode)

        if (props.status !== status) {
            const response = await profileAPI.updateUserStatus(status)

            if (response.resultCode === 0) {
                dispatch(actions.setUserStatus(status))
            }
        }
    }

    const statusChange = (e: ChangeEvent<HTMLInputElement>) => {
        setStatus(e.currentTarget.value)
    }

    return (
        <div>
            {props.editingAbility ?
                editMode ?
                    <input type="text" onBlur={toggleEditMode} onChange={statusChange} value={status} autoFocus/>
                    : <span className={`${s.status} ${s.status_editingAbility}`}
                            onDoubleClick={toggleEditMode}>{props.status}</span>
                : <span className={s.status}>{props.status}</span>
            }
        </div>
    )
}


export default StatusWithHooks