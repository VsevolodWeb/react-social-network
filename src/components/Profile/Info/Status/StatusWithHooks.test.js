import React from "react";
import { create } from "react-test-renderer"
import StatusWithHooks from "./StatusWithHooks";



describe("Status component", () => {
    const component = create(<StatusWithHooks status="test status"/>);
    it("status should be created successfully", () => {
        expect(component.toJSON()).toMatchSnapshot();
    })

    it("test", () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
});