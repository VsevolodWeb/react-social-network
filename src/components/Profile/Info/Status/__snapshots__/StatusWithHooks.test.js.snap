// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component status should be created successfully 1`] = `
<div>
  <span
    className="status"
  >
    test status
  </span>
</div>
`;

exports[`Status component test 1`] = `
<div>
  <span
    className="status"
  >
    test status
  </span>
</div>
`;
